public/app/plugins/datasource/loki/configuration/DerivedField.tsx

```diff
@@ -60,7 +60,7 @@ export const DerivedField = (props: Props) => {
   };
 
   return (
-    <div className={className}>
+    <div className={className} data-testid="derived-field">
```

Comment:
is there a way to avoid using a `data-testid` attribute? in general, we should test for things the user might see on the screen (https://testing-library.com/docs/queries/about#priority ) , so, if possible, could we for example test for the remove-field-button or something like that?

Reply:
I actually added it because I didn't find anything more representative of a `DerivedField` component, which is what I want to find and count.

I understand where you're comming from, but the priotity docs you're pointing me to doesn't apply here, since I'm not trying to select (or interact with) a button or an input. The intention of the library's author is to motivate developers to interact with DOM in a more accessible way, for example, by using button's labels instead of adding a test-id to it.

Here, I'm just trying to know if there is a `DerivedField` component or not, and how many are there. Whether there is a test-id or not doesn't have any effect on accesibility or user interactions.

For example, if this component had a title, I would gladly use something like a `getByText("Derived Field");`. Instead, I have a bunch of fields with labels "Name" or "Regex", which I'm not super comfortable using to assert if there is a component or not, etc.

Let me know if what I'm saying is not clear.

Reply:
For example, this pseudo-code:

```javascript
test("It renders as many Derived Fields as passed", () => {
  render(<DerivedFields fields=[//two fields] />);
  // ...
  expect(screen.getByText("Label")).toHaveLength(2); // ??
});
```

This is what I would like to circumvent here.

Reply:
We also aim to make tests resilient to change (as much as possible), so my goal here is that the DerivedFields (plural) renders as many DerivedField as it should. If at some point `Label` becomes `Name` or `Regex` becomes `Regular Expression`, this test will fail, even if the underlying implementation did not change.

Reply:
i see your point. i must admit, it still feels a little like we are testing an internal detail of `DerivedFields` (how many `DerivedField` objects are inside), and i would love to be able to test something more "from the outside" property there. but i understand why it is done this way. 👍

Reply:
Thanks! Yes, it's definitely not ideal, but I tried to find the best option for that, even though it's potentially a nitpicky situation.